sqlglot/dialects/presto.py

```diff
@@ -57,8 +57,14 @@ def _no_sort_array(self: Presto.Generator, expression: exp.SortArray) -> str:
 
 def _schema_sql(self: Presto.Generator, expression: exp.Schema) -> str:
     if isinstance(expression.parent, exp.PartitionedByProperty):
-        columns = ", ".join(f"'{c.name}'" for c in expression.expressions)
-        return f"ARRAY[{columns}]"
+        column_literals = []
+        for c in expression.expressions:
+            if isinstance(c, exp.Func):
+                column_literals.append(self.sql(c))
+            else:
+                column_literals.append(self.sql(c, "this"))
+
+        return self.sql(exp.Array(expressions=[exp.Literal.string(c) for c in column_literals]))
```

Comment:
Is this right? Why are we converting everything into a string at the end? wouldn't that convert function calls, which have already been generated, to string literals within the array?

Reply:
Ok sounds good, thanks for checking 👍